latch_cli/nextflow/workflow.py

```diff
@@ -0,0 +1,293 @@
+from dataclasses import fields, is_dataclass
+from enum import Enum
+from pathlib import Path
+from textwrap import dedent
+from typing import Any, List, Optional, Tuple
+
+import click
+
+import latch.types.metadata as metadata
+from latch.types.directory import LatchDir, LatchOutputDir
+from latch.types.file import LatchFile
+from latch_cli.snakemake.config.utils import get_preamble, type_repr
+from latch_cli.snakemake.utils import reindent
+from latch_cli.utils import identifier_from_str, urljoins
+
+template = """\
+from dataclasses import dataclass
+from enum import Enum
+import os
+import subprocess
+import requests
+import shutil
+from pathlib import Path
+import typing
+import typing_extensions
+
+from latch.resources.workflow import workflow
+from latch.resources.tasks import nextflow_runtime_task, custom_task
+from latch.types.file import LatchFile
+from latch.types.directory import LatchDir, LatchOutputDir
+from latch.ldata.path import LPath
+from latch_cli.nextflow.workflow import get_flag
+from latch_cli.nextflow.utils import _get_execution_name
+from latch_cli.utils import urljoins
+from latch.types import metadata
+from flytekit.core.annotation import FlyteAnnotation
+
+import latch_metadata
+
+@custom_task(cpu=0.25, memory=0.5, storage_gib=1)
+def initialize() -> str:
+    token = os.environ.get("FLYTE_INTERNAL_EXECUTION_ID")
+    if token is None:
+        raise RuntimeError("failed to get execution token")
+
+    headers = {{"Authorization": f"Latch-Execution-Token {{token}}"}}
+
+    print("Provisioning shared storage volume... ", end="")
+    resp = requests.post(
+        "http://nf-dispatcher-service.flyte.svc.cluster.local/provision-storage",
+        headers=headers,
+        json={{
+            "storage_gib": {storage_gib},
+        }}
+    )
+    resp.raise_for_status()
+    print("Done.")
+
+    return resp.json()["name"]
+
+
+{preambles}
+
+{samplesheet_funs}
+
+@nextflow_runtime_task(cpu={cpu}, memory={memory})
+def nextflow_runtime(pvc_name: str, {param_signature}) -> None:
+    try:
+        shared_dir = Path("/nf-workdir")
+
+{samplesheet_constructors}
+
+        shutil.copytree(
+            Path("/root"),
+            shared_dir,
+            ignore=lambda src, names: ["latch", ".latch", "nextflow", ".nextflow", "work", "results"],
```

Comment:
would also add common packages like `conda` etc to ignored list

Reply:
it would be odd if conda was installed in the root directory no? I think stuff like venv we might want to just keep because they could be used by the nextflow tasks.

Reply:
conda installs in the root dir by default